File: rb/features.rb
# PublibikeStations SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module PublibikeStationsFeatures
  def self.make_feature(name)
    case name
    when "base"
      PublibikeStationsBaseFeature.new
    when "test"
      PublibikeStationsTestFeature.new
    else
      PublibikeStationsBaseFeature.new
    end
  end
end

File: py/features.py
# PublibikeStations SDK feature factory

from feature.base_feature import PublibikeStationsBaseFeature
from feature.test_feature import PublibikeStationsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PublibikeStationsBaseFeature(),
        "test": lambda: PublibikeStationsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
